### PR TITLE
Pre-Publish Checklist: Update strings

### LINKED ISSUE #5848: Update Pre-publish checklist to use the revised text strings
Pre-publish checklist updated text strings suggested in collaboration with Huge: https://docs.google.com/spreadsheets/d/1PjaRp4bueQ3Cv14A7I696m0vjGCIX_kN9tIGh-V-Bkc/edit?ts=5fdcc25f#gid=920793050

Once implemented, we should update the file to make the WIP tab the new official strings tab.

### PR DESCRIPTION
**Pull request by samwhale**
## Summary

Updates the text in the pre-publish checklist. Using updated strings from [Huge suggestion document](https://docs.google.com/spreadsheets/d/1PjaRp4bueQ3Cv14A7I696m0vjGCIX_kN9tIGh-V-Bkc/edit?ts=5fdcc25f#gid=920793050). Sorry, this one is tedious to test.

## Relevant Technical Choices

none

## To-do

Once implemented, we should update the file to make the WIP tab the new official strings tab.

## User-facing changes

Pre publish checklist has updated text

## Testing Instructions

- trigger warnings or errors in the pre-publish checklist to see the new text

|Before|After|
|--|--|
|<image src="https://user-images.githubusercontent.com/22185279/104783448-465f2d00-5743-11eb-8c9f-0b80e0f56881.png" width="250px" />|<image src="https://user-images.githubusercontent.com/22185279/104783476-5545df80-5743-11eb-8863-2389a99d0a3d.png" width="250px" />|

---

Fixes #5848

### CODE REVIEW

File: assets/src/edit-story/app/prepublish/constants.js [hunk 1]
```diff
@@ -53,208 +53,256 @@ export const PRE_PUBLISH_MESSAGE_TYPES = {
...
+        <TranslateWithMarkup>
+          {sprintf(
+            /* translators: 1: minimum story title length in words. 2: maximum story title length in characters. */
+            __(
+              "- Keep under %1$d words<br/>- Don't exceed %2$d characters",
+              'web-stories'
+            ),
+            MAX_STORY_TITLE_LENGTH_WORDS,
+            MAX_STORY_TITLE_LENGTH_CHARS
+          )}
```
Comment (swissspidy) [inline assets/src/edit-story/app/prepublish/constants.js]: First, these hyphens everywhere should really not be part of the translatable strings itself. They should be really be done in CSS, for example by actually using `li` items (because it's a list of recommendations after all) and using `list-style-type: '-'`. It does not make sense for translators to translate a hyphen.

Second, this one should be split up into two and use `_n()` instead of `__()`. That's because in some languages, the plurals of `words` and `characters` needs to be translated differently depending on the number being referenced.

Example:

```js
sprintf(
  _n( 'Keep under %d word', 'Keep under %d words', MAX_STORY_TITLE_LENGTH_WORDS, 'web-stories' ),
  MAX_STORY_TITLE_LENGTH_WORDS
);

sprintf(
  _n( 'Don't exceed %d character', 'Don't exceed %d characters', MAX_STORY_TITLE_LENGTH_CHARS, 'web-stories' ),
  MAX_STORY_TITLE_LENGTH_CHARS
);
```

So basically like this:

```js
<ul>
	  <li>
		    {sprintf(
		      _n( 'Keep under %d word', 'Keep under %d words', MAX_STORY_TITLE_LENGTH_WORDS, 'web-stories' ),
		      MAX_STORY_TITLE_LENGTH_WORDS
		    )}
	  </li>
	  <li>
		    {sprintf(
		      _n( 'Don't exceed %d character', 'Don't exceed %d characters', MAX_STORY_TITLE_LENGTH_CHARS, 'web-stories' ),
		      MAX_STORY_TITLE_LENGTH_CHARS
		    )}
	  </li>
</ul>
```

Comment (swissspidy) [reply assets/src/edit-story/app/prepublish/constants.js]: The same goes for all other similar changes  here  of course

Comment (samwhale) [reply assets/src/edit-story/app/prepublish/constants.js]: Good shout! I'll get these fixed 👍

File: assets/src/edit-story/app/prepublish/constants.js [hunk 2]
```diff
@@ -432,9 +432,9 @@ export const MESSAGES = {
           </li>
           <li>
             {sprintf(
-              /* translators: %d: maximum percentage of characters, depending on number of pages. */
+              /* translators: 1: maximum percentage of characters, depending on number of pages. */
               __(
-                'Keep the total number of text-heavy pages under %d% of total number of story pages',
+                'Keep the total number of text-heavy pages under %1$d%% of total number of story pages',
```
Comment (swissspidy) [inline assets/src/edit-story/app/prepublish/constants.js]: If there's only one placeholder, just using `%d` instead of numbered placeholders (`%1$d`) is preferred.

Still need the dobule `%%` to get a single percentage sign of course. Alternatively do `MAX_CHARACTER_PERCENTAGE + '%'` below

File: assets/src/edit-story/app/prepublish/constants.js [hunk 3]
```diff
@@ -44,6 +48,14 @@ const MIN_VIDEO_FPS = 24;
 const MAX_VIDEO_LENGTH_SECONDS = 60;
 const MAX_VIDEO_LENGTH_MINUTES = Math.floor(MAX_VIDEO_LENGTH_SECONDS / 60);
 
+const DashedList = styled.ul`
+  list-style-type: '-';
+
+  li {
+    padding-left: 5px;
```
Comment (BrittanyIRL) [inline assets/src/edit-story/app/prepublish/constants.js]: nit: let's do 4px because design wants everything to be divisible by 4.

File: assets/src/edit-story/app/prepublish/constants.js [hunk 4]
```diff
@@ -53,276 +65,517 @@ export const PRE_PUBLISH_MESSAGE_TYPES = {
...
-      MAIN_TEXT: __('Video missing description', 'web-stories'),
-      HELPER_TEXT: __(
-        'Add a video description. Descriptions help with indexability and accessibility (for videos without captions). Include any burned-in text inside of the video.',
-        'web-stories'
+      MAIN_TEXT: __('Add video description', 'web-stories'),
+      HELPER_TEXT: (
+        <DashedList>
+          <li>
+            {__(
+              'Improves indexability and accessibility (for videos without captions)<br/>- Include any text or captions that are already a rendered part of the video',
```
Comment (BrittanyIRL) [inline assets/src/edit-story/app/prepublish/constants.js]: instead of a `<br />` tag and a dash, could you move the second phrase starting with `Include` to be a second list item?

File: assets/src/edit-story/app/prepublish/constants.js [hunk 5]
```diff
@@ -53,276 +65,517 @@ export const PRE_PUBLISH_MESSAGE_TYPES = {
...
+                  <a
+                    href="https://amp.dev/documentation/guides-and-tutorials/start/create_successful_stories/#visual-treat"
+                    rel="noreferrer"
+                    target="_blank"
+                  />
+                ),
+              }}
+            >
+              {__(
+                '<a>Use full screen videos and images whenever possible to create a more immersive experience. (more info)</a>',
```
Comment (BrittanyIRL) [inline assets/src/edit-story/app/prepublish/constants.js]: nit: this is the only piece of text with a period at the end of the list item. Thoughts on removing it for consistency?

**Review by BrittanyIRL — COMMENTED** [review]
The string updates look great. 

This isn't about the strings, i'm sorry for suggesting added scope, but now that the helper text is separated as list items (which are alot easier to read), i think adding some padding-left to the `ul` that holds the helper text list items to line up the list items dash with the start of the parent main text might be nice. If that's adding a lot of time, don't worry about it and we can catch it later. We don't want to spend a lot of energy styling this unnecessarily since we'll need to update it anyway here soon for adding in the new designs.

File: assets/src/edit-story/app/prepublish/constants.js (hunk 4 above)
Comment (samwhale) [reply assets/src/edit-story/app/prepublish/constants.js]: good catch!

**Comment by samwhale** [comment]
Updated checklist looks like:
<image src="https://user-images.githubusercontent.com/22185279/105102975-fd71e600-5a6c-11eb-9d06-8d8d379e2586.png" width="200px" />

File: assets/src/edit-story/app/prepublish/constants.js [hunk 6]
```diff
@@ -482,7 +489,7 @@ export const MESSAGES = {
               }}
             >
               {__(
-                '<a>Use full screen videos and images whenever possible to create a more immersive experience. (more info)</a>',
+                '<a>Use full screen videos and images whenever possible to create a more immersive experience (more info)</a>',
```
Comment (swissspidy) [inline assets/src/edit-story/app/prepublish/constants.js]: If the whole string is wrapped in a link, move the link outside the `__()`. Doesn't make to translate it in that case. Also means you won't need `TranslateWithMarkup`

Comment (samwhale) [reply assets/src/edit-story/app/prepublish/constants.js]: Ah nice catch!

**Review by embarks — APPROVED** [review]
All the strings look good. The only feedback I have is, some of the unit tests just use the strings from the constants file so that they don't need to be updated when the strings get updated and some of them use a string which has to be updated to match the string in the constants file. Might reduce maintenance in the future to make the tests consistently use the same strings from the constants file, but it's not necessary if that takes a lot of time to fix the tests and they pass now.

**Comment by samwhale** [comment]
Great idea!

I think for now we'll keep going with this since this pr is pretty large, but I can add a tech debt ticket to make these tests more resilient

File: assets/src/edit-story/app/prepublish/constants.js [hunk 7]
```diff
@@ -53,276 +62,516 @@ export const PRE_PUBLISH_MESSAGE_TYPES = {
...
-        `${COVER_DIMENSION_WIDTH_PX}x${COVER_DIMENSION_HEIGHT_PX}px`,
-        `${ASPECT_RATIO_LEFT}:${ASPECT_RATIO_RIGHT}`
+      MAIN_TEXT: __('Add story cover image', 'web-stories'),
+      HELPER_TEXT: (
+        <DashedList>
+          <li>{__('Use as a representation of the story', 'web-stories')}</li>
+          <li>{__('Avoid images with embedded text', 'web-stories')}</li>
+          <li>
+            {sprintf(
+              /* translators: %s: minimum cover dimension width x minimum cover dimension height.  */
```
Comment (swissspidy) [inline assets/src/edit-story/app/prepublish/constants.js:103]:
```suggestion
              /* translators: %s: image dimensions, 640x853px.  */
```

File: assets/src/edit-story/app/prepublish/constants.js [hunk 8]
```diff
@@ -53,276 +62,516 @@ export const PRE_PUBLISH_MESSAGE_TYPES = {
...
-          'web-stories'
-        ),
-        `${PUBLISHER_LOGO_DIMENSION}x${PUBLISHER_LOGO_DIMENSION}px`,
-        `${PUBLISHER_LOGO_RATIO}x${PUBLISHER_LOGO_RATIO}px`
+      MAIN_TEXT: __('Upload a bigger logo', 'web-stories'),
+      HELPER_TEXT: (
+        <DashedList>
+          <li>
+            {sprintf(
+              /* translators: %s: minimum publisher logo dimensions. */
```
Comment (swissspidy) [inline assets/src/edit-story/app/prepublish/constants.js]:
```suggestion
              /* translators: %s: image dimensions. */
```

File: assets/src/edit-story/app/prepublish/constants.js [hunk 9]
```diff
@@ -53,276 +62,516 @@ export const PRE_PUBLISH_MESSAGE_TYPES = {
...
-        `${ASPECT_RATIO_LEFT}:${ASPECT_RATIO_RIGHT}`
+      MAIN_TEXT: __(
+        'Choose an image with the correct aspect ratio',
+        'web-stories'
+      ),
+      HELPER_TEXT: (
+        <DashedList>
+          <li>
+            {sprintf(
+              /* translators: %s: minimum cover dimension width X minimum cover dimension height.  */
```
Comment (swissspidy) [inline assets/src/edit-story/app/prepublish/constants.js]:
```suggestion
              /* translators: %s: image dimensions.  */
```

File: assets/src/edit-story/app/prepublish/constants.js [hunk 10]
```diff
@@ -53,276 +62,516 @@ export const PRE_PUBLISH_MESSAGE_TYPES = {
...
-        `${ASPECT_RATIO_LEFT}:${ASPECT_RATIO_RIGHT}`
+      MAIN_TEXT: __(
+        'Choose a bigger image for your story cover',
+        'web-stories'
+      ),
+      HELPER_TEXT: (
+        <DashedList>
+          <li>
+            {sprintf(
+              /* translators: %s: minimum cover dimension width X minimum cover dimension height.  */
```
Comment (swissspidy) [inline assets/src/edit-story/app/prepublish/constants.js]:
```suggestion
              /* translators: %s: image dimensions. */
```

File: assets/src/edit-story/app/prepublish/constants.js [hunk 11]
```diff
@@ -53,276 +62,516 @@ export const PRE_PUBLISH_MESSAGE_TYPES = {
...
+          <li>
+            {sprintf(
+              /* translators: %s: minimum publisher logo dimensions. */
+              __("Use an image that's at least %s", 'web-stories'),
+              `${PUBLISHER_LOGO_DIMENSION}x${PUBLISHER_LOGO_DIMENSION}px`
+            )}
+          </li>
+          <li>
+            {sprintf(
+              /* translators: %s: publisher logo dimension ratio. */
```
Comment (BrittanyIRL) [inline assets/src/edit-story/app/prepublish/constants.js]: probably image dimensions here too rather than publisher logo dimensions for consistency with changes pascal suggested.

**Review by BrittanyIRL — APPROVED** [review]
Thanks for the updates! Agree with Pascal's i18n suggestions.

File: assets/src/edit-story/app/prepublish/constants.js (hunk 11 above)
Comment (swissspidy) [reply assets/src/edit-story/app/prepublish/constants.js]: To give some context to my suggestions, merely as an FYI:

When a string is presented to a translator in the UI they're going to see the string as well as any additional translator comment added to it.

It therefore makes sense to use the same translator comment for all instances of the same string, otherwise you'll see the string with three different translator comments added to it.

**Review by embarks — APPROVED** [review]

File: assets/src/edit-story/app/prepublish/constants.js (hunk 11 above)
Comment (samwhale) [reply assets/src/edit-story/app/prepublish/constants.js]: Ah makes sense. Thank you!
